lndr-backend/src/Lndr/Server.hs

```diff
@@ -239,8 +241,23 @@ verifyIndividualRecord creditHash = do
     let recordNotFound = throwError $
             err404 { errBody = "Credit hash does not refer to pending bilateral settlement record" }
     bilateralCreditRecord <- maybe recordNotFound pure recordM
-    verifySettlementPayment bilateralCreditRecord
+    let txHashNotFound = throwError $
+            err404 { errBody = "Bilateral settlement record does not have a transaction hash" }
+    transHash <- maybe txHashNotFound pure $ txHash bilateralCreditRecord
+    multiSettlementRecords <- liftIO $ withResource pool $ Db.lookupCreditsByTxHash transHash
+    let settlementAmount = foldr getSettlementAmount 0 multiSettlementRecords
+    liftIO $ pushLogStrLn loggerSet . toLogStr $ ("SETTLEMENT AMOUNT TO VERIFY " :: String) ++ (show settlementAmount)
+    verifySettlementPayment bilateralCreditRecord settlementAmount
     liftIO $ withResource pool $ Db.verifyCreditByHash creditHash
     web3Result <- finalizeTransaction configTVar bilateralCreditRecord
                         `catchError` (pure . T.pack . show)
     liftIO $ pushLogStrLn loggerSet . toLogStr . ("WEB3: " ++) . show $ web3Result
+
+
+getSettlementAmount :: BilateralCreditRecord -> Integer -> Integer
+getSettlementAmount (BilateralCreditRecord creditRecord _ _ _) cumulativeSettlementAmt = 
+    cumulativeSettlementAmt + ( getAmount $ settlementAmount creditRecord )
+
+getAmount :: Maybe Integer -> Integer
+getAmount (Just amt) = amt
+getAmount Nothing = 0
```

Comment:
delete this `getAmount` function. The `Data.Maybe` module contains a standard function that does this called `fromMaybe :: a -> Maybe a -> a`. http://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Maybe.html